### PR TITLE
PRM-632 Added config value for PEGA BearerToken

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -81,4 +81,12 @@ class AppConfig @Inject()(config: Configuration, servicesConfig: ServicesConfig)
     agentClientLookupHost +
       servicesConfig.getConfString(vatAgentClientLookupFrontendStartUrl, "") +
       s"?redirectUrl=${agentClientLookupRedirectUrl(uri)}"
+
+  lazy val pegaBearerToken: String = "bearerToken" //TODO: bearerToken to be configured for PEGA
```
Comment (LewisMitchell) [inline app/config/AppConfig.scala:85]: Similar to the layout above, we tend to add the key to application.conf and assign a value to it. So in application.conf have a new key-value:
```
pega {
    bearerToken = "placeholder"
}
```

Then you can replace this with `config.get[String]("pega.bearerToken")`

File: app/config/AppConfig.scala [hunk 2]
```diff
@@ -81,4 +81,12 @@ class AppConfig @Inject()(config: Configuration, servicesConfig: ServicesConfig)
...
 }
+
+trait AppConfigTrait{
```
Comment (LewisMitchell) [inline app/config/AppConfig.scala:88]: I don't think we need this trait.

File: test/config/MockAppConfigTrait.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import org.scalamock.handlers.CallHandler
+import org.scalamock.scalatest.MockFactory
+
+trait MockAppConfigTrait extends MockFactory {
```
Comment (LewisMitchell) [inline test/config/MockAppConfigTrait.scala:22]: There is a simpler solution to this, see comments in HeaderGeneratorSpec

File: test/connectors/HeaderGeneratorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+class HeaderGeneratorSpec extends WordSpec with Matchers with HeaderNames with MockAppConfigTrait {
+
+  val testHeaderGenerator = new HeaderGenerator(mockAppConfig)
+  val bearerToken           = "someToken"
+
+  "HeaderGenerator.headersForPEGA" must {
+
+      "include the Authorization header if it is supplied in AppConfig" in {
+      MockAppConfig.pegaBearerToken returns bearerToken
```
Comment (LewisMitchell) [inline test/connectors/HeaderGeneratorSpec.scala:33]: You can mock AppConfig:
`val mockAppConfig: AppConfig = mock(classOf[AppConfig])`

Then when your method calls AppConfig - you mock the response:
`when(mockAppConfig.pegaBearerToken).thenReturn(bearerToken)`

File: test/connectors/HeaderGeneratorSpec.scala [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      implicit val hc: HeaderCarrier = HeaderCarrier()
+      val headers                    = testHeaderGenerator.headersForPEGA().toMap
+
+      headers(AUTHORIZATION) shouldBe s"Bearer $bearerToken"
+    }
+
+    "not include the Authorization header if it is empty string in AppConfig" in {
+      MockAppConfig.pegaBearerToken returns ""
```
Comment (LewisMitchell) [inline test/connectors/HeaderGeneratorSpec.scala:41]: Same as above
`when(mockAppConfig.pegaBearerToken).thenReturn("")`

File: test/connectors/MockHeaderGenerator.scala [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+package connectors
+
+import uk.gov.hmrc.http.HeaderCarrier
+import org.scalamock.handlers.CallHandler
+import org.scalamock.scalatest.MockFactory
+
+trait MockHeaderGenerator extends MockFactory{
```
Comment (LewisMitchell) [inline test/connectors/MockHeaderGenerator.scala:23]: I don't think this is being used?

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -22,6 +22,7 @@ object AppDependencies {
     "com.vladsch.flexmark"     %  "flexmark-all"              % "0.36.8"           % "test, it",
     "com.github.tomakehurst"  %  "wiremock-jre8"            % "2.26.3"           % "it",
     "uk.gov.hmrc"             %% "service-integration-test" % "1.1.0-play-28"    % "test, it",
-    "org.mockito"             % "mockito-core"              % "3.4.6"            % "test, it"
+    "org.mockito"             % "mockito-core"              % "3.4.6"            % "test, it",
+    "org.scalamock"           %% "scalamock-scalatest-support" % "3.6.0"         % "test"
```
Comment (LewisMitchell) [inline project/AppDependencies.scala:26]: If you action the other comments, I don't think we need this library.

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just some changes to make 😄

File: app/config/AppConfig.scala (hunk 1 above)
Comment (ruthdas03) [reply app/config/AppConfig.scala:85]: Ok, then it seems the bearer token is not getting it from generated from any Authentication Server.

Comment (LewisMitchell) [reply app/config/AppConfig.scala:85]: The bearer token obviously won't change anyway. We will be provided with the bearer token later so we can pass it with each request. By having it in application.conf makes it easier to change the bearer token without having to release a new version of the service.

**Comment by ruthdas03** [comment]
> Looks good, just some changes to make 😄

Made the changes.

**Review by LewisMitchell — APPROVED** [review]
LGTM.

**Pull request merged**